use_cases/refresher.go

```diff
@@ -30,26 +32,18 @@ func NewRefresher(reader reader, saver saver, fetcher fetcher) Refresher {
 }
 
 func (r Refresher) Refresh(ctx context.Context) error {
-	pokemons, err := r.Read()
-	if err != nil {
-		return err
-	}
+	var pokemons []models.Pokemon
 
-	for i, p := range pokemons {
-		urls := strings.Split(p.FlatAbilityURLs, "|")
-		var abilities []string
-		for _, url := range urls {
-			ability, err := r.FetchAbility(url)
-			if err != nil {
-				return err
-			}
+	dataChan := readData(r)
 
-			for _, ee := range ability.EffectEntries {
-				abilities = append(abilities, ee.Effect)
-			}
-		}
+	// Workers
+	pokemonWorker1 := getAbilities(dataChan, r)
```

Comment:
Is there a way to produce workers dynamically instead of declaring them statically?, Why three workers?